### ISSUE TITLE
SI Fixes request: Batlin wearing the Fellowship Medallion when viewed through the eye of the moon

### ISSUE THREAD

**Issue by DominusExult** [exult/exult#435]
When Batlin is viewed through the Eye of the Moon and speaks to his henchmen the Batlin face with the Fellowship Medallion is used.
Chronologically this happens AFTER he has lost/left his medallion in Shamino's castle, so he shouldn't be wearing it.

This is a bug in the original usecode (?). Teleport to dec 1463,997 to view the Eye of the Moon.

![image](https://github.com/exult/exult/assets/3313896/b964383a-7f65-495f-a659-0b740c5f4488)

**Comment by KnightCaptainU7** [comment]
Yes, it's a Origin Usecode bug. Sadly the cutscene is buried in the code tied to the shape of Batlin, and it's not the only occurrence within the file of the wrong frame being used.

I last edited the file in 2017, and never finished the conversion.
[batlin.uc.txt](https://github.com/exult/exult/files/14120271/batlin.uc.txt)

`void shapeBatlin shape#(0x13E) () // 318

if (gflags[BEGIN_VISION_OF_BATLIN] && ((event == PROXIMITY) && (!gflags[FINISHED_VISION_OF_BATLIN])))
		{
if (UI_get_npc_id(item) == 5)
			{
				UI_init_conversation();
				// Should use frame 1 here, without the medallion.
				UI_show_npc_face0(FACE_BATLIN, 1);
				say("@To thee I give the anchor's job... The Avatar must not enter the Shrine of Balance before the Wall of Lights is fully open. Thou must stop the Avatar, should the others fail.@");`

**Comment by KnightCaptainU7** [comment]
I wonder if simply changing the order of the two face frames would be a better fix? Then it would be correct. And we may assume Batlin removed it before his battle with Cantra rather than after?

**Comment by KnightCaptainU7** [comment]
[287_Batlin_face_order_fixed.zip](https://github.com/exult/exult/files/14391454/287_Batlin_face_order_fixed.zip)

**Comment by KnightCaptainU7** [comment]
![image](https://github.com/exult/exult/assets/117867844/389d7538-c976-40b6-a90d-69aba99064c0)

**Comment by DominusExult** [comment]
Your screenshot shows the henchmen scene, I'm talking about the scene when he opens the Wall of Light. The game uses the correct face (without medallion) and just switching the faces around will have it look wrong then. Kind of the same in the dream land, though in the timeline a face without medallion would fit but not from a view that he would surely dream of himself with a medallion. Hence the question if that would be easier to usecode patch. We could do both face frames 0 and 1 to be without medallion and add a third frame with and point that scene at that frame.

> This sort of thing seems like it would be easiest to fix if we had some mechanism to binary patch the original usecode instead of adding new usecode that can call into the original.

Yes, but at the same time, it's really just nitpicks :)

**Comment by DominusExult** [comment]
I've got it somewhat working....
Somewhat as in all the conversations at that scene run twice 🤣
(once with the medallion followed by once without)

[batlin.uc.txt](https://github.com/user-attachments/files/26010161/batlin.uc.txt)

(I'm using these in the global flags declarations to be somewhat synced with Marzo's newest global flags names:
```
enum Batlin_visions {
	DOING_SPINEBREAKER_VISION = 0x24F,
	DID_SPINEBREAKER_VISION = 0x250
};
```
)
